### PR TITLE
Static Code Analysis with Php Inspections (EA Extended)

### PR DESCRIPTION
**Pull request by kalessil**
Couple of findings

### CODE REVIEW

File: lib/Doctrine/ORM/Tools/Console/Command/ConvertDoctrine1SchemaCommand.php [hunk 1]
```diff
@@ -180,6 +180,7 @@ public function convertDoctrine1Schema(array $fromPaths, $destPath, $toType, $nu
                 );
             }
         }
+        unset($dirName);
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Tools/Console/Command/ConvertDoctrine1SchemaCommand.php]: Not sure if I like this inspection. `$dirName` is not used after this location: let the stack garbage collection do its job.

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php [hunk 2]
```diff
@@ -1629,6 +1629,7 @@ protected function _validateAndCompleteOneToOneMapping(array $mapping)
                     ? $joinColumn['fieldName']
                     : $joinColumn['name'];
             }
+            unset($joinColumn);
```
Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php]: Same as below: this inspection is (IMO) not a good idea.

**Review by Ocramius — CHANGES_REQUESTED** [review]

File: lib/Doctrine/ORM/Tools/Console/Command/ConvertDoctrine1SchemaCommand.php (hunk 1 above)
Comment (kalessil) [inline lib/Doctrine/ORM/Tools/Console/Command/ConvertDoctrine1SchemaCommand.php]: It's more tricky: https://3v4l.org/QKOKH - here unset should clean the scope after foreach.

Comment (Ocramius) [inline lib/Doctrine/ORM/Tools/Console/Command/ConvertDoctrine1SchemaCommand.php]: I know why this is applied, it just doesn't make sense if the variables are unused. I'd rather prefer extracting the loop into a private method instead.

Comment (kalessil) [inline lib/Doctrine/ORM/Tools/Console/Command/ConvertDoctrine1SchemaCommand.php]: I see, good approach btw. Reverting the change.

File: lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php (hunk 2 above)
Comment (kalessil) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php]: Reverting. I also curious why it's not a good idea, why?

Comment (Ocramius) [inline lib/Doctrine/ORM/Mapping/ClassMetadataInfo.php]: It's just an additional instruction (also to read - not worried about performance)

**Comment by kalessil** [comment]
@Ocramius : I reverted usset constructs here, but checks still saying that you requested changes.

**Comment by Ocramius** [comment]
@kalessil thanks for this, merging!

**Pull request merged**